widgets/refinement-list.js

```diff
@@ -51,17 +57,30 @@ function refinementList({
     throw new Error(usage);
   }
 
+  if (templates !== defaultTemplates) {
+    templates = Object.assign({}, defaultTemplates, templates);
+  }
+
   return {
     getConfiguration: () => ({
       [operator === 'and' ? 'facets' : 'disjunctiveFacets']: [facetName]
     }),
     render: function({results, helper}) {
+      var values = results.getFacetValues(facetName, {sortBy: sortBy}).slice(0, limit);
+
+      if (values.length === 0) {
```

Comment:
As I said this refactor will go in another PR (where I will add the header/footer to every widget)